src/api.py

```diff
@@ -64,73 +64,66 @@ def storage_urls():
-                    else:
-                        matches: dict = f.search_metadata(
-                            user_search["blob_type"], user_search, session
-                        )
-                    return Response(body=matches).json()
-            except Exception as e:
+    @app.get("/blob")
+    def read_blob(request: Request):
+        try:
+            # user_search = request.query_params
+            user_search = request.query_params
+            if not "blob_type" in user_search.keys():
+                return Response(
+                    status_code=400, error_message="Must provide blob_type"
+                ).json()
+            if not f.validate_search_fields(user_search):
                 return Response(
-                    status_code=500, error_message=f"UnexpectedError: {e}"
-                ).json
+                    status_code=400,
+                    error_message="KeyError: invalid blob_type or search field",
+                ).json()
+            # blob_type only - return all entries for blob_type
+            with Session() as session:
+                if len(user_search) == 1:
+                    matches = f.all_entries(user_search["blob_type"], session)
+                    return Response(body=matches).json()
+                else:
+                    matches: dict = f.search_metadata(
+                        user_search["blob_type"], user_search, session
+                    )
+                return Response(body=matches).json()
+        except Exception as e:
+            return Response(status_code=500, error_message=f"UnexpectedError: {e}").json
 
-        elif request.method == "POST":
-            try:
-                new_blob_info = BlobPostSchema.parse_obj(request.get_json())
-                blob_type = new_blob_info.metadata["blob_type"]
-                parsed_metadata = BLOB_TYPES[blob_type].parse_obj(
-                    new_blob_info.metadata
+    @app.post("/blob")
+    def create_blob(new_blob_info: BlobPostSchema):
+        try:
+            blob_type = new_blob_info.metadata["blob_type"]
+            parsed_metadata = BLOB_TYPES[blob_type].parse_obj(new_blob_info.metadata)
+            # add paths to blob table (id = hash, path = blobs/blob_type/hash)
+            with Session() as session:
+                paths_added = f.add_blob_paths(
+                    parsed_metadata.blob_hash, new_blob_info.paths, session
                 )
-                # add paths to blob table (id = hash, path = blobs/blob_type/hash)
-                with Session() as session:
-                    paths_added = f.add_blob_paths(
-                        parsed_metadata.blob_hash, new_blob_info.paths, session
+                if not paths_added:
+                    return Response(
+                        status_code=500, error_message="Error adding paths"
+                    ).json()
+                # add metadata entry to db
+                if new_blob_info.new == NEW_BLOB:
```

Comment:
i am not 100% what `.new` property refers to, but my guess is you want a term like "lifecycle".

Reply:
refers to status of blob user is trying to save: is it a New_Blob (not in cabinet yet), a New_location (in cabinet but not yet saved at requested location) or is it False (user trying to submit duplicate)